### PR TITLE
Feature/9 show route on the map

### PR DESCRIPTION
**Pull request by Rodrigo-Schar**
-added UI and draw route
-fix maps key
-merge route list by category
-added show route from list
-merge route list fix
-fix model
-change icons
-fix model conflict
-fix request changes
-merge develop conflicts
-fix google key

### CODE REVIEW

File: app/src/main/java/com/jalasoft/routesapp/ui/routes/view/RouteSelectedFragment.kt [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        if (isLocationPermissionGranted()) {
+            getLocation()
+        }
+
+        val start = GoogleMapsHelper.locationToLatLng(route?.start!!)
+        val end = GoogleMapsHelper.locationToLatLng(route?.end!!)
+        mMap!!.addMarker(MarkerOptions().position(start).title(R.string.start_of_route.toString()).icon(GoogleMapsHelper.bitmapFromVector(requireContext(), R.drawable.ic_start_route)))
+        mMap!!.addMarker(MarkerOptions().position(end).title(R.string.end_of_route.toString()).icon(GoogleMapsHelper.bitmapFromVector(requireContext(), R.drawable.ic_end_route)))
+
+        addStopMarkers(route!!.stops)
```
Comment (jmacboy) [inline app/src/main/java/com/jalasoft/routesapp/ui/routes/view/RouteSelectedFragment.kt]: never force unwrap variables, validate if this route and nmap are different than null first

File: app/src/main/java/com/jalasoft/routesapp/ui/routes/view/RouteSelectedFragment.kt [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+                MarkerOptions().position(GoogleMapsHelper.locationToLatLng(i)).title(R.string.bus_stop.toString()).icon(GoogleMapsHelper.bitmapFromVector(requireContext(), R.drawable.ic_bus_stop))
+            )
+        }
+    }
+
+    private fun drawPolyline(list: List<Location>) {
+        for (i in list.indices) {
+            val item = list[i]
+            if (item == list.last()) {
+                mMap!!.addPolyline(
```
Comment (jmacboy) [inline app/src/main/java/com/jalasoft/routesapp/ui/routes/view/RouteSelectedFragment.kt]: same here with the force unwrapping

File: app/src/main/java/com/jalasoft/routesapp/ui/routes/view/RouteSelectedFragment.kt [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+                    PolylineOptions()
+                        .add(GoogleMapsHelper.locationToLatLng(list[i - 1]), GoogleMapsHelper.locationToLatLng(item))
+                        .width(10f)
+                        .color(resources.getColor(R.color.color_primary, null))
+                        .geodesic(true)
+                )
+            } else {
+                mMap!!.addPolyline(
```
Comment (jmacboy) [inline app/src/main/java/com/jalasoft/routesapp/ui/routes/view/RouteSelectedFragment.kt]: same here with the force unwrapping

**Review by YawarValeriano — APPROVED** [review]
Seems good

File: app/src/main/java/com/jalasoft/routesapp/util/helpers/GoogleMapsHelper.kt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            vectorDrawable.intrinsicHeight,
+            Bitmap.Config.ARGB_8888
+        )
+
+        // below line is use to add bitmap in our canvas.
+        val canvas = Canvas(bitmap)
+
+        // below line is use to draw our
+        // vector drawable in canvas.
+        vectorDrawable.draw(canvas)
```
Comment (jmacboy) [inline app/src/main/java/com/jalasoft/routesapp/util/helpers/GoogleMapsHelper.kt]: no need for all these "below line" comments

**Review by jmacboy — CHANGES_REQUESTED** [review]

**Review by jmacboy — APPROVED** [review]

**Pull request merged**